src/libraries/Microsoft.Extensions.Logging.EventSource/src/EventSourceLogger.cs

```diff
@@ -57,11 +57,11 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             {
                 return;
             }
+            string message = formatter(state, exception);
```

Comment:
@tarekgh looks like this was related to your feedback.  Don't we normally try to avoid work and allocations when the event source is disabled?

Reply:
I was going to say we should have the string variable but set it in the branch and only compute it when needed.

Reply:
I think this is can be defined as `message = null` and then we initialize it later when it is needed and it is null.

Reply:
yah